src/win/pipe.c

```diff
@@ -46,6 +46,11 @@ struct uv__ipc_queue_item_s {
 /* A zero-size buffer for use by uv_pipe_read */
 static char uv_zero_[] = "";
 
+#ifdef UV_USE_PIPE_INTERRUPTER
+/* To prevent deadlocks we will interrupt synchronous pipe read every 2.5 s. */
```

Comment:
Can you add some more details here? Tell the story so anyone browsing the code won't be surprised.